Source/Core/Common/QoSSession.h

```diff
@@ -0,0 +1,72 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <enet/enet.h>
+
+namespace Common
+{
+class QoSSession
+{
+public:
+  // 1 0 1 1 1 0 0 0
+  // DSCP      ECN
+  constexpr static int ef_tos = 0b10111000;
+
+  QoSSession() = default;
+  ~QoSSession();
+
+  QoSSession(ENetPeer* peer, int tos_val = ef_tos);
+
+  QoSSession& operator=(const QoSSession&) = delete;
+  QoSSession(const QoSSession&) = delete;
+
+  QoSSession& operator=(QoSSession&& session)
+  {
+#if defined(_WIN32)
```

Comment:
Generally checking for self-assignment in the copy assignment/move assignment operators is done:

```cpp
if (this != &session)
{
  // move or copy stuff...
}

return *this;
```